### DISCUSSION TITLE
What triggers the creation of a new "expense account"?

### DISCUSSION THREAD

**Discussion by wsw70** [firefly-iii/firefly-iii#9207] [Firefly III]
I import operations from a few banks and one of them automatically creates "expense accounts" 

![image](https://github.com/user-attachments/assets/9c69197a-7b02-42ab-b16a-a81be19cbfe9)

I am not sure yet if this is a good or bad thing<sup>*</sup> but I would like to understand **what in an operation triggers the creation** of this "expense account"

---
<sup>*</sup>they are not strictly speaking accounts, but on the other hand it may be easier to categorize them, I don't know yet)

**Comment by JC5** [comment]
Looking at your import, could it be that your import doesn't have the necessary information, or perhaps not mapped correctly?

**Reply by wsw70** [reply]
I am importing via GoCardless so I have no control over the format

**Reply by JC5** [reply]
In that case, there isn't much I can do. If the destination account information isn't present, it can't be used to create a new account. If it's possible, you can use a rule to create the destination account based on the notes or the description.

**Reply by wsw70** [reply]
> In that case, there isn't much I can do. If the destination account information isn't present, it can't be used to create a new account. If it's possible, you can use a rule to create the destination account based on the notes or the description.

Oh yes, this is clear and I did not mean to imply this. My question was rather about the data that is used to create the destination account. I downloaded a CSV export of the "better" bank and it way richer than the one from the "less good bank". My only hope is to put pressure to provide more information (which will certainly fail)

I was considering rules but 

- for known destinations: they need to be set one by one and this will take a long time
- for unknown/new destinations: I would need to create more rules

I think I would rather go for a script that will load the operations from the "less good bank" and process them through a regex (it will not be ideal, but better that I have now and can make exceptions). I am not sure if such a script is "pluggable" into firefly but I will have a closer look at the docs.
